src/app/suplier/suplier-suggestion-details/suplier-suggestion-details.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SuplierModel } from 'src/models/suplier';
+import { SuplierService } from '../suplier.service';
+
+@Component({
+  selector: 'app-suplier-suggestion-details',
+  templateUrl: './suplier-suggestion-details.component.html',
+  styleUrls: ['./suplier-suggestion-details.component.css']
+})
+export class SuplierSuggestionDetailsComponent implements OnInit {
+  suplier: SuplierModel | undefined;
+  lstSuplier: any;
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private suplierService: SuplierService
+              ) {
+
+            }
+
+  ngOnInit(): void {
+    this.route.params.subscribe(
+      params => {
+          const id = +params['id'];
+          this.getSuplier(id);
+      }
+    );
+  }
+
+  viewSuplier(id: number) {
+    this.router.navigate([`/suplier-suggestion/${id}`]);
+  }
+
+  getSuplier(_productId: number) {
```

Comment:
change the name getSupplierById or getById 
_productId change the id name to id or supplierId